src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -766,7 +766,15 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                                     return;
                                 }
 
-                                xmlStreamOpt.SetLength(0);
+                                try
+                                {
+                                    xmlStreamOpt.SetLength(0);
+                                }
+                                catch (Exception e)
+                                {
+                                    MessageProvider.ReportStreamWriteException(e, finalXmlFilePath, diagnostics);
```

Comment:
What side effects is the rest of the code going to hit if the stream doesn't have a length of 0?

Reply:
Setting the length to zero clears the file, so you could end up with garbage at the end of the file if the new file is shorter than the old one.

Reply:
I think I missed the return when I looked at this at first, and thus though we'd continue with xml stream being the wrong length.